File: domain/src/main/java/com/viewnext/domain/repository/GetDetallesRepository.java
package com.viewnext.domain.repository;

import com.viewnext.domain.model.Detalles;
import java.util.List;

/**
 * Repositorio para obtener los detalles de la instalación o entidad correspondiente.
 * Define los métodos para recuperar y refrescar los detalles desde la fuente de datos.
 */
public interface GetDetallesRepository {
    /**
     * Obtiene los detalles almacenados localmente.
     * @return Lista de detalles
     */
    List<Detalles> getDetalles();

    /**
     * Refresca los detalles desde la fuente de datos.
     * @param detallesCallback Callback que será notificado cuando la operación haya finalizado.
     */
    void refreshDetalles(DetallesCallback<List<Detalles>> detallesCallback);
}

File: domain/src/main/java/com/viewnext/domain/repository/GetDetallesRepository.kt
package com.viewnext.domain.repository

import com.viewnext.domain.model.Detalles

/**
 * Repositorio para obtener los detalles de la instalación o entidad correspondiente.
 * Define los métodos para recuperar y refrescar los detalles desde la fuente de datos.
 */
interface GetDetallesRepository {
    /**
     * Obtiene los detalles almacenados localmente.
     * @return Lista de detalles
     */
    val detalles: MutableList<Detalles?>?

    /**
     * Refresca los detalles desde la fuente de datos.
     * @param detallesCallback Callback que será notificado cuando la operación haya finalizado.
     */
    fun refreshDetalles(detallesCallback: DetallesCallback<MutableList<Detalles?>?>?)
}
